definitions/types/aws-arn.json

```diff
@@ -7,7 +7,7 @@
     "arn:aws:rds::ACCOUNT_NUMBER:db/prod",
     "arn:aws:ec2::ACCOUNT_NUMBER:vpc/vpc-foo"
   ],
-  "pattern": "^arn:aws:[a-zA-Z0-9._-]*:[a-zA-Z0-9._-]*:[0-9]{12}:[a-zA-Z0-9._-]+(?:[\/:][a-zA-Z0-9\/._-]+)?$",
+  "pattern": "^arn:aws:[a-zA-Z0-9._-]*:[a-zA-Z0-9._-]*:(?:[0-9]{12})?:[a-zA-Z0-9._-]+(?:[\/:][a-zA-Z0-9\/._-]+)?$",
```

Comment:
this change allows 0 OR 12 digits in the account id space of the ARN by making an optional non-capturing group expecting exactly 12 digits.

Reply:
Perfect!